app/models/trip_searcher.rb

```diff
@@ -0,0 +1,56 @@
+class TripSearcher
+  include ActiveModel::SerializerSupport
+
+  DEFAULT_PER_PAGE = 20
+
+  attr_reader :per_page, :page
+
+  def initialize(params)
+    @params = params
+    @per_page = positive_or_default(params[:per_page].to_i, DEFAULT_PER_PAGE)
+    @page = positive_or_default(params[:page].to_i, 1)
+  end
+
+  def results
+    @results ||= paginated_results
+  end
+
+  def total_results
+    filtered_results.count
+  end
+
+  def total_pages
+    total_results / per_page
+  end
+
+  def offset
+    per_page * (page - 1)
+  end
+
+  def valid?
+    [
+      @params[:route_id].present?,
+    ].any?
+  end
+
+  private
+
+  def positive_or_default(v, d)
```

Comment:
We should extract this into a Searchable concern or something similar. We've got a lot of duplication between this and the StopSearcher that could probably be put in there as well.